test-e2e/translation-api.js

```diff
@@ -142,6 +142,97 @@ test('translation api - put() and get() fields', async (t) => {
   }
 })
 
+test('translation api - passing `lang` to dataType', async (t) => {
+  const [manager] = await createManagers(1, t, 'mobile')
+  const project = await manager.getProject(
+    await manager.createProject({
+      configPath: defaultConfigPath,
+    })
+  )
+
+  // fields
+  const fields = await project.field.getMany()
+  const fieldTranslationsDoc = fields
+    .map((field) => {
+      const matchingTranslation = fieldTranslations.find((translation) => {
+        return translation.message === fieldsTranslationMap[field.label]
+      })
+      if (matchingTranslation)
+        return { docIdRef: field.docId, ...matchingTranslation }
+    })
+    .filter(isDefined)
+  for (const translationDoc of fieldTranslationsDoc) {
+    const { docIdRef, message, fieldRef } = await project.$translation.put(
+      translationDoc
+    )
+    const translatedField = await project.field.getByDocId(docIdRef, {
+      lang: 'es',
+    })
+
+    t.is(
+      translatedField[fieldRef],
+      message,
+      `passing 'lang' returns the correct translated field`
+    )
+    const untranslatedField = await project.field.getByDocId(docIdRef)
+    t.not(
+      untranslatedField[fieldRef],
```

Comment:
Any ts incantation that would allow me to use the `fieldRef` as key without complains @EvanHahn ??

Reply:
Done in #619.